libs/shared/ui-components/src/Video/Video.tsx

```diff
@@ -0,0 +1,131 @@
+import { useState } from 'react';
+
+import clsx from 'clsx';
+import ReactPlayer from 'react-player';
+
+import { createMarkup } from '@quansight/shared/utils';
+
+import { VideoThumbnail } from './VideoThumbnail';
+import { TVideoProps } from './types';
+
+const playerOptions = {
+  autoPlay: 0,
+  disablekb: 1,
+  showinfo: 1,
+  volume: 0,
+  muted: 0,
+  pip: 1,
+  controls: 2,
+  enablejsapi: 1,
+};
+
+const videoSizeStyles: Record<TVideoProps['size'], string> = {
+  small: 'lg:w-1/2 lg:min-w-[50%] lg:pt-[28.125%]',
+  medium: 'lg:w-3/4 lg:min-w-[75%] lg:pt-[42.18%]',
+  large: '',
+};
+
+const alignVideoStyles: Record<TVideoProps['align'], string> = {
+  left: 'lg:flex-row',
+  right: 'lg:flex-row-reverse',
+  center: '',
+};
+
+const topSpaceStyles: Record<TVideoProps['topSpace'], string> = {
+  none: 'mt-0',
+  small: 'mt-[4rem] xl:mt-[8rem]',
+  large: 'mt-[6rem] xl:mt-[10rem]',
+};
+
+const bottomSpaceStyles: Record<TVideoProps['bottomSpace'], string> = {
+  none: 'mb-0',
+  small: 'mb-[4rem] xl:mb-[8rem]',
+  large: 'mb-[6rem] xl:mb-[10rem]',
+};
+
+const textStyles =
+  'my-[1.6rem] text-[1.6rem] text-black marker:text-black prose sm:text-[1.8rem] sm:my-[4rem] xl:leading-[2.7rem]';
+
+export const Video = ({
+  url,
+  thumbnail,
+  size,
+  align,
+  title,
+  sideText,
+  text,
+  topSpace,
+  bottomSpace,
+}: TVideoProps) => {
+  const [playing, setPlaying] = useState(false);
+
+  const handleToggleVideoPlay = () => setPlaying((prevState) => !prevState);
+
+  if (!url) {
+    return null;
+  }
+
+  const playerConfig = {
+    vimeo: { playerOptions },
+    youtube: { playerVars: playerOptions },
+  };
+
+  const styles = {
+    header: `
+      mb-[5.6rem] text-[4rem] font-extrabold leading-[4.9rem] text-center font-heading text-violet
+      sm:mx-auto sm:max-w-[70rem] sm:text-[4.8rem] sm:leading-[5.3rem]
+      xl:mb-[6.8rem]
+    `,
+    player: clsx(
+      'relative pt-[56.25%] w-full',
+      videoSizeStyles[size || 'large'],
+    ),
+    text: clsx(textStyles, 'hidden', size === 'small' && 'lg:block'),
```

Comment:
So is this always going to be hidden? Am I reading this right?

Reply:
It's `an optional rich text field that allows text to be placed in the open space to the right and left of the video, respectively` from the requirements so it will be visible if chosen size is 'small' on desktop view. So at the moment:
- if video size is other than 'small' - left/right settings are not applicable and side text is not showing,
- if video size is small - left/right settings are applicable on desktop and side text is showing on desktop - on the mobile there is full width view and side text is hidden.

Please confirm if this requirement was understood properly or we need to change something here.

Reply:
Sounds good @bskinn